plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -68,6 +67,8 @@
   private static final String SERVICE_KEY_ENVIRONMENT_VARIABLE =
       "GOOGLE_APPLICATION_CREDENTIALS"; //$NON-NLS-1$
 
+  private final Image gcpIcon = SharedImages.GCP_IMAGE_DESCRIPTOR.createImage();
```

Comment:
This approach isn't a good practice as the instance may be created but never used (and hence the image is created unnecessarily), and there's no guarantee that the instance will be created on the SWT thread.  Better would be for `getImage()` to create the image lazily and `dispose()` to do a check-if-null.

Reply:
OK, that makes sense. BTW, the image doesn't need to be created on the UI thread.